src/test/java/org/jabref/gui/UpdateTimestampListenerTest.java

```diff
@@ -0,0 +1,96 @@
+package org.jabref.gui;
+
+import java.util.Optional;
+
+import org.jabref.logic.preferences.TimestampPreferences;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class UpdateTimestampListenerTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private BibDatabase database;
+
+    @Before
+    public void setUp(){
+        database = new BibDatabase();
+    }
+
+    @Test
+    public void updateTimestampEnabled(){
+        final String TIMESTAMP_FIELD = "timestamp";
+        final String BASE_DATE = "2000-1-1";
+        final String NEW_DATE = "2000-1-2";
+
+        final boolean INCLUDE_TIMESTAMP = true;
+
+        JabRefPreferences preferencesMock = mock(JabRefPreferences.class);
```

Comment:
As mentionend above, when you directly pass the TimestampPreferences Object/Mock, you don't need to have a dependency on JabRef Preferences

Reply:
Furthermore, you can move the timestamp constants to private static final in the class body. 
 object creation and add the when(xxx) methods to the setup method. Then you can avoid such code duplication

Reply:
Will do this when moving the initialization not dependent on the defined constants into the setup method.

I would, however, prefer to not make the instance variable "private static final". I would like to just make it "private" and instantiate a new mock object for both tests in the setup method instead.
Even though old "when" calls on methods should be overwritten by newer "when" calls (according to the mockito javadoc), using new objects avoids confusion when reading the tests and removes the possibility of dependencies or interactions between the tests.

Would that also be acceptable?

Reply:
The setup() method is called before every test

Reply:
Maybe I was not clear enough when explaining my point of view.
I don't want to have just one mock object for TimestampPreferences that is shared between all tests (which would be the case if TimestampPreferences were made "private static final"). I would like to use private instance variables that are instantiated in the setup method instead. This would mean that we would have a different mock object for each test case.

I am about to push a commit which moves some of the initialization to the setup method. Then we have the code to better discuss the issue and see if any other changes are warranted.